fix flaky test TestRunRenewCommands

sometimes when running all the kubeadm unit tests with this command from kubernetes/kubernetes:
```
go test ./cmd/kubeadm/app/... -count=1
```

there is one test that fails from time to time:
```
Error: unknown command "args --cert-dir=/tmp/1668299040" for "apiserver"
Usage:
  apiserver [flags]

Flags:
      --cert-dir string     The path where to save the certificates (default "/etc/kubernetes/pki")
      --config string       Path to a kubeadm configuration file.
  -h, --help                help for apiserver
      --kubeconfig string   The kubeconfig file to use when talking to the cluster. If the flag is not set, a set of standard locations can be searched for an existing kubeconfig file. (default "/etc/kubernetes/admin.conf")

--- FAIL: TestRunRenewCommands (0.84s)
    --- FAIL: TestRunRenewCommands/apiserver (0.04s)
        certs_test.go:285: certificate apiserver was not renewed as expected
```

--- a/cmd/kubeadm/app/cmd/certs_test.go
+++ b/cmd/kubeadm/app/cmd/certs_test.go
@@ -23,6 +23,7 @@ import (
 	"crypto"
 	"crypto/x509"
 	"fmt"
+	"io"
 	"os"
 	"path/filepath"
 	"strings"
@@ -269,7 +270,7 @@ func TestRunRenewCommands(t *testing.T) {
 			if len(test.Args) > 0 {
 				args = test.Args + " " + args
 			}
-			err := cmdtestutil.RunSubCommand(t, renewCmds, test.command, args)
+			err := cmdtestutil.RunSubCommand(t, renewCmds, test.command, io.Discard, args)
 			// certs renew doesn't support positional Args
 			if (err != nil) != test.expectedError {
 				t.Errorf("failed to run renew commands, expected error: %t, actual error: %v", test.expectedError, err)

--- a/cmd/kubeadm/test/cmd/util.go
+++ b/cmd/kubeadm/test/cmd/util.go
@@ -18,6 +18,7 @@ package kubeadm
 
 import (
 	"bytes"
+	"io"
 	"os"
 	"os/exec"
 	"testing"
@@ -50,8 +51,9 @@ func RunCmd(command string, args ...string) (string, string, int, error) {
 }
 
 // RunSubCommand is a utility function for kubeadm testing that executes a Cobra sub command
-func RunSubCommand(t *testing.T, subCmds []*cobra.Command, command string, args ...string) error {
+func RunSubCommand(t *testing.T, subCmds []*cobra.Command, command string, output io.Writer, args ...string) error {
 	subCmd := getSubCommand(t, subCmds, command)
+	subCmd.SetOut(output)
 	subCmd.SetArgs(args)
 	if err := subCmd.Execute(); err != nil {
 		return err

